sphinx_gallery/gen_rst.py

```diff
@@ -592,10 +588,12 @@ def execute_script(script_blocks, script_vars, gallery_conf):
         # for more details.
         sys.argv[0] = script_vars['src_file']
         sys.argv[1:] = []
+        gc.collect()
```

Comment:
@larsoner sorry this PR is very old but will this clean up `global_variables` ?

Reply:
I think that was the goal, yeah

Reply:
Thanks! So running gc.collect in a resetter would be redundant then?

Reply:
I'd have to think about execution order, but in principle I think so yes. But also running it again shouldn't hurt. Personally I do `gc.collect` in a resetter in MNE-Python then check that there are no objects of specific types lingering around, e.g.:

https://github.com/mne-tools/mne-python/blob/66c40caea1fd343fbe49a34815542065ceecdf3d/doc/conf.py#L478-L491

Reply:
Thanks! I ask because I noticed in MNE you run `gc.collect` in the resetter and again in `_assert_no_instances`, but I guess that is because `_assert_no_instances` is run outside of the resetter. Anyway thank you!